### PR TITLE
Remove max LED number constraint from Matrix layout

### LINKED ISSUE #1804: WLED matrix layout of 64x64 is not saved
- [x] I confirm that this is an issue rather than a question.

## Bug report
I have smaller setups of 16x16 , 32x16 that seems to be working fine with Hyperion  . WLED  has expanded the supported number of leds as well as the led panels types , with hub75 being added , I am testing with a small 64x64 panel that works perfectly fine when used with internal effects as well as ddp streaming from xlights . WLED at that moment can work with around 128x128 

#### Steps to reproduce
Setup 64x64 matrix in Hyperion 

#### What is expected?
64x64 matrix layout to be saved 

#### What is actually happening?
error below and when streaming the output is only on half of the 64x64 and mirrored 
Image

### PR DESCRIPTION
**Pull request by Lord-Grey**
**Summary**

### Changed
- Removed maximum LED number constraint from Matrix layout schema which was not synced with the UI behaviour (#1804)
- Streamlined call-back code chain

### Added
- JSON-API - Limit  update emission frequency: Images (25Hz), raw LED-Colors (40Hz) & LED Device data (200Hz) 
- Effects - Limit the maximum update rate to 200Hz

**What kind of change does this PR introduce?** (check at least one)

- [X] Bugfix
- [X] Feature
- [ ] Code style update
- [X] Refactor
- [ ] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of web configuration, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing setups:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's body (e.g. `Fixes: #xxx[,#xxx]`, where "xxx" is the issue number)

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated (docs/docs/en)
- [ ] Related tests have been updated

**PLEASE DON'T FORGET TO ADD YOUR CHANGES TO CHANGELOG.MD**
- [ ] Yes, CHANGELOG.md is also updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**
Fixes #1804

### CODE REVIEW

**Comment by dosipod** [comment]
As listed in the issue #1804 for 64x64 layout not being saved , this is solved now and I could save 64x64 hub75 panel .

Hyperion app performance  is a bit slow and the app itself is sluggish ( I expected that )  .In xlights the ui performance  was okay but effects have the same issues with  DDP  so not all effects work as expected but this is still testing on wifi s3 board and wled side is still wip ( Changes are planned for streaming but  may be not going to be available until Jan 2025 ) and by that time we are hoping to get wired boards .

I will continue to test performance  with Hyperion considering the above two limitation ( WLED and Network ) as you wish if we
are to update this PR or open a discussion for the same  . For the time being the minor issue I see that is not allowing us to test in a correct way and might be you could help with clarify if we are doing a wrong setting ( We have done the same setting on other smaller setup ) that makes the effects mirrored and only on half of the 64x64 . Due to things being slow ,below is a small quick and dirty test to compare with a calibration image and result from xlights and Hyperion . The same result is in all Hyperion effects and screen capture so everything is mirrored and only on half of the matrix .As most effects in Hyperion are symmetrical it is a bit hard to show on picture or video but we could also do that if you wish  but I hope it is clear .

As said we are willing to go deep in that if you see it to be of use at this time else will not report until Jan 2025 once wled is update and our new HW is hopefully on site ,  still I will be testing further to at least get more familiar with the ui which I must admit I am not very used to specially with large matrix .

image

**Comment by Lord-Grey** [comment]
@dosipod Thanks for the feedback. Let us work jointly to see how to get it working.

> As most effects in Hyperion are symmetrical it is a bit hard to show on picture or video but we could also do that if you wish but I hope it is clear .

I would suggest you run the "LED Test -Sequence" effect.
That lights up LEDs in an RGB sequence starting from LED #0 to the total amount.
From there you can identify how the numbering is applied to your matrix.

image

To see the way numbering just use a smaller number of LEDs
That numbering must match the WLED one.

image

>  a calibration image 

How did you feed the calibration image?
Via the Picture effect? 

image

Would you mind sharing the calibration image?

**Comment by dosipod** [comment]
Something I want to do is reset the whole Hyperion setup and delete everything to only add the 64x64 again , what is the best way to do that

**Comment by Lord-Grey** [comment]
> Something I want to do is reset the whole Hyperion setup and delete everything to only add the 64x64 again , what is the best way to do that

Delete the configuration database….

**Comment by dosipod** [comment]
Very good but from where exactly meaning where does Hyperion store its config 
I have removed the files from ~/Library/Preferences/  but still the config is not reset

And I can not see anything under   ~/Library/Application Support/  so it could be removed 

image

**Comment by dosipod** [comment]
In the docs it is listing .hyperion so this might be it

**Comment by dosipod** [comment]
@Lord-Grey @Paulchen-Panther  I just reset the config and only changed the led cabling to be parallel and everything worked now and performance is not really bad even better then xlights   so excellent 

image

I will shortly post a test of a video of screen capture ( this is the ultimate stress test for us but if you have anything in mind the do tell )

**Comment by dosipod** [comment]
Could not load a proper video due to size and lag from my wifi capture cam , but you get the picture as there is no lag on wifi for the matrix so it is very impressive
https://github.com/user-attachments/assets/5324229a-a781-4d0d-9345-e63c15991c34
You may delete the video if you like

**Comment by Lord-Grey** [comment]
@dosipod Great that it is working now and indeed the output looks impressive!

**Comment by dosipod** [comment]
@Lord-Grey I think there is two point to that , 1st from wled side this might be an issue unless an S3 with octal PSRAM 
is used and 2nd point is using a wired board . Unfortunately at the moment I do not have Ethernet board
with S3 and octal PSRAM nor the number of panels to try but I think it should be fine giving the above 
result on wifi which I only tried with a classic esp32 and HD-WF2 ( S3 without PSRAM so its as good as esp32 )

File: .github/workflows/qt5_6.yml [hunk 1]
```diff
@@ -120,22 +120,10 @@ jobs:
...
-            python3 --version
-          echo '::endgroup::'
-
-          echo '::group::Set up Xcode'
-            # Xcode 14.2 is the latest version that can run on macOS 12
-            sudo xcode-select -s /Applications/Xcode_14.2.app/Contents/Developer
-            xcode-select -p
-            xcodebuild -version
-          echo '::endgroup::'
+      - name: Temporarily downgrade CMake to 3.28.3 # Please remove if GitHub has updated Cmake (greater than 3.30.0)
```
Comment (Paulchen-Panther) [inline .github/workflows/qt5_6.yml]: Der Downgrade ist nicht mehr notwendig.

Comment (Lord-Grey) [reply .github/workflows/qt5_6.yml]: HaHa der Downgrade ist gar nicht Teil des finalen PRs :)
Können wir aber mit dem nächsten PR berücksichtigen....

**Pull request merged**